src/Polysemy/ConstraintAbsorber/MonadError.hs

```diff
@@ -0,0 +1,52 @@
+{-# LANGUAGE FlexibleInstances           #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving  #-}
+{-# LANGUAGE MultiParamTypeClasses       #-}
+{-# LANGUAGE ScopedTypeVariables         #-}
+{-# LANGUAGE UndecidableInstances        #-}
+{-# OPTIONS_GHC -fplugin=Polysemy.Plugin #-}
+
+module Polysemy.ConstraintAbsorber.MonadError
+  (
+    absorbError
+  )
+where
+
+import           Polysemy
+import           Polysemy.ConstraintAbsorber
+import           Polysemy.Error
+import qualified Control.Monad.Error.Class as S
+
+------------------------------------------------------------------------------
+-- | absorb a @MonadError e@ constraint into @Member (Error e) r => Sem r@
+absorbError :: Member (Error e) r
+  => (S.MonadError e (Sem r) => Sem r a) -> Sem r a
+absorbError = absorbWithSem @(S.MonadError _) @Action
+  (ErrorDict throw catch)
+  (Sub Dict) 
+{-# INLINEABLE absorbError #-}
+
+-- | A dictionary of the functions we need to supply
+-- to make an instance of Error
+data ErrorDict e m = ErrorDict
+  {
```

Comment:
Nit: here and elsewhere, prefer to put the first element on the same line as the opener, eg:

```haskell
data Foo = Foo
  { throwError_ :: ..
```

likewise for module export lists